### PR TITLE
chore: example with clerk

### PR DESCRIPTION
**Pull request by Mr0Bread**
Minimal example. If user is not signed in - uploading is prohibited

- [ ] Add examples with `pages` dir

### CODE REVIEW

File: examples/with-clerk/next.config.mjs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+  /**
+   * If you are using `appDir` then you must comment the below `i18n` config out.
+   *
+   * @see https://github.com/vercel/next.js/issues/41980
+   */
+  i18n: {
+    locales: ["en"],
+    defaultLocale: "en",
+  },
```
Comment (juliusmarminge) [inline examples/with-clerk/next.config.mjs:19]: rm

**Comment by Mr0Bread** [comment]
Will add example with pages dir

File: examples/with-clerk/src/server/uploadthing.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    },
+  })
+    .middleware(({ req }) => {
+      const { userId } = auth();
+
+      if (!userId) {
+        throw new Error("Please sign in");
+      }
+
+      return {};
```
Comment (juliusmarminge) [inline examples/with-clerk/src/server/uploadthing.ts:29]:
```suggestion
      return { userId };
```

**Review by juliusmarminge — APPROVED** [review]

**Comment by Mr0Bread** [comment]
@juliusmarminge I see that you approved the changes, but can I ask you to rename the example to `with-appdir-and-clerk`

@markflorkowski wants to have similar but for pages directory

**Comment by juliusmarminge** [comment]
This looks solid - the pages/ example can be a separate PR

**Pull request merged**